gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py

```diff
@@ -0,0 +1,183 @@
+
+
+def main(args):
+    """
+    Script used to get variants impacted by homalt hotfix.
+    Most of the work was done in notebooks, and any results created in notebooks are noted with comments.
+    """
+
+    hl.init(log="/get_impacted_variants.log", default_reference="GRCh38")
+
+    logger.info("Reading sparse MT and metadata table with release only samples...")
+    mt = get_gnomad_v3_mt(
+        key_by_locus_and_alleles=True, samples_meta=True, release_only=True,
+    ).select_entries(*SPARSE_ENTRIES)
+
+    if args.test:
+        logger.info("Filtering to two partitions on chr20")
+        mt = hl.filter_intervals(mt, [hl.parse_locus_interval("chr20:1-1000000")])
+        mt = mt._filter_partitions(range(2))
+
+    logger.info(
+        "Adding annotation for whether original genotype (pre-splitting multiallelics) is het nonref..."
+    )
+    # Adding a Boolean for whether a sample had a heterozygous non-reference genotype
+    # Need to add this prior to splitting MT to make sure these genotypes
+    # are not adjusted by the homalt hotfix downstream
+    mt = mt.annotate_entries(het_non_ref=mt.LGT.is_het_non_ref())
+    # TODO: Need to fix main code to add this annotation and use it also, where to add?
+
+    logger.info("Splitting multiallelics...")
+    mt = hl.experimental.sparse_split_multi(mt, filter_changed_loci=True)
+
+    logger.info("Reading in v3.0 release HT (to get frequency information)...")
+    freq_ht = release_sites().versions["3"].ht().select_globals().select("freq")
+
+    logger.info("Reading in info HT...")
+    info_ht = get_info().ht()
+
+    logger.info(
+        "Checking for variants with het nonref calls that were incorrectly adjusted with homalt hotfix..."
+    )
+    sites_ht = get_het_non_ref_impacted_var(mt, info_ht, freq_ht)
+
+    logger.info("Densifying MT to het non ref sites only...")
+    mt = densify_sites(
+        mt, sites_ht, last_END_position.versions["3.1"].ht(), semi_join_rows=False,
+    )
+    mt = mt.filter_rows(
+        (hl.len(mt.alleles) > 1) & (hl.is_defined(sites_ht[mt.row_key]))
+    )
+
+    logger.info(
+        "Writing out dense MT with only het nonref calls that may have been incorrectly adjusted with the homalt hotfix..."
+    )
+    if args.test:
+        mt = mt.checkpoint(
+            "gs://gnomad-tmp/release_3.1.2/het_nonref_fix_sites_3.1.2_test.mt",
+            overwrite=True,
+        )
+    else:
+        mt = mt.checkpoint(
+            "gs://gnomad-tmp/release_3.1.2/het_nonref_fix_sites.mt", overwrite=True
+        )
+
+    logger.info(
+        "Writing out variants with het nonref calls that may have been incorrectly adjusted with the homalt hotfix..."
+    )
+    if args.test:
+        mt.rows().write(
+            "gs://gnomad-tmp/release_3.1.2/het_nonref_fix_sites_3.1.2_test.ht",
+            overwrite=True,
+        )
+    else:
+        mt.rows().write(
+            "gs://gnomad/release/3.1.2/ht/het_nonref_fix_sites.ht", overwrite=True
```

Comment:
I would need to do the filter step on 135 two times then right? I don't even think I really need this HT honestly, I really only care about the densified MT and the frequency HTs will contain this site information so I might just remove this.

Reply:
Actually, I think I will keep as is because this will ensure that I have a saved the HT of the sites where I recalculated frequency on the densified MT